### PR TITLE
Permalink updates, dependency updates, and example phrasing cleanup

### PR DESCRIPTION
**Pull request by gerth2**
Updated examples to use github permalinks, rather than branch-latest raw literal includes. Updated line numbers to match.

Updated some of the phrasing and explanations in the examples to trim down total length in an attempt to make it more approachable.

Re-ran `pip freeze > requirements.txt` on win11 python 3.10 to capture a missing sphinx dependency.

### CODE REVIEW

**Comment by gerth2** [comment]
Addresses https://github.com/PhotonVision/photonvision-docs/issues/166

**Comment by gerth2** [comment]
Supersedes https://github.com/PhotonVision/photonvision-docs/pull/165

File: requirements.txt [hunk 1]
```diff
@@ -7,4 +7,5 @@ sphinxcontrib-ghcontributors
 sphinx-notfound-page
 sphinxext-remoteliteralinclude
 sphinx_design==0.0.13
+sphinx-rtd-theme==1.0.0
```
Comment (mdurrani808) [inline requirements.txt:10]: What is this used for? To my knowledge, we don't use the RTD theme anywhere.

**Review by mdurrani808 — APPROVED** [review]
Looks good! Thank you for this.

**Pull request merged**